go/runtime/history/history.go

```diff
@@ -134,22 +138,41 @@ func (h *runtimeHistory) RuntimeID() common.Namespace {
 	return h.runtimeID
 }
 
-func (h *runtimeHistory) Commit(blk *roothash.AnnotatedBlock, roundResults *roothash.RoundResults, notify bool) error {
-	err := h.db.commit(blk, roundResults)
+func (h *runtimeHistory) CommitBatch(blks []*roothash.AnnotatedBlock, results []*roothash.RoundResults, notify bool) error {
+	if len(blks) == 0 && len(results) == 0 {
+		return nil
+	}
+
+	err := h.db.commitBatch(blks, results)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to commit batch: %w", err)
 	}
 
 	// Notify the pruner what the new round is.
-	h.pruneCh.In() <- blk.Block.Header.Round
+	lastBlk := blks[len(blks)-1]
+	h.pruneCh.In() <- lastBlk.Block.Header.Round
 
-	// If no local storage worker, notify the block watcher that new block is committed,
+	// If no local storage worker, notify the block watcher about new blocks,
 	// otherwise the storage-sync-checkpoint will do the notification.
 	if h.hasLocalStorage || !notify {
 		return nil
 	}
-	h.blocksNotifier.Broadcast(blk)
+	for _, blk := range blks {
+		h.blocksNotifier.Broadcast(blk)
+	}
+
+	return nil
+}
 
+func (h *runtimeHistory) Commit(blk *roothash.AnnotatedBlock, roundResults *roothash.RoundResults, notify bool) error {
```

Comment:
I would move `Commit` before `CommitBatch`, as is simpler and to be in line with the interface.